src/System.IO.Pipelines/tests/FlushAsyncCancellationTests.cs

```diff
@@ -327,82 +328,25 @@ public async Task FlushAsyncThrowsIfPassedCanceledCancellationTokenAndPipeIsAble
         }
 
         [Fact]
-        public void ReadAsyncCompletesIfFlushAsyncCanceledMidFlush()
+        public async Task ReadAsyncReturnsDataAfterCancelledReadWasCompleted()
```

Comment:
The test and the test name seem to be mismatched. After cancelling the read, you initiated a new read (which when completed returned the expected data). I might be mistaken, but the cancelled read didn't complete and return the data. Can you please clarify?

Reply:
Scenario I'm trying to test is:

1. Read is initiated and goes async
2. Read is cancelled
3. Data is written into the pipe before GetResult() was called on awaiter

CancelledReadWasCompleted is about compleating cancelled reader by writing more data.

Before the fix this scenario caused second ReadAsync to never return.